Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Fleet/Driver.hs

```diff
@@ -2504,3 +2508,54 @@ getListOfDriversMultiFleet _ mbDriverPhNo fleetOwnerIds _ mbIsActive mbLimit mbO
   driverAssociationAndInfo <- FDV.findAllActiveDriverByFleetOwnerIdWithDriverInfoMF fleetOwnerIds limit offset mobileNumberHash mbName mbSearchString mbIsActive mode
   let (fleetDriverAssociation, person, driverInformation) = unzip3 driverAssociationAndInfo
   return (fleetDriverAssociation, person, driverInformation)
+
+postDriverFleetUnlinkDocument ::
+  ShortId DM.Merchant ->
+  Context.City ->
+  Text ->
+  Id Common.Driver ->
+  CommonDriver.DocumentType ->
+  Maybe Text ->
+  Flow APISuccess
+postDriverFleetUnlinkDocument merchantShortId opCity requestorId driverId documentType mbFleetOwnerId = do
+  requestedPerson <- QPerson.findById (Id requestorId) >>= fromMaybeM (PersonDoesNotExist requestorId)
+  (entityRole, entityId) <- validateRequestorRoleAndGetEntityId requestedPerson mbFleetOwnerId
+  merchant <- findMerchantByShortId merchantShortId
+  merchantOpCityId <- CQMOC.getMerchantOpCityId Nothing merchant (Just opCity)
+  let personId = cast @Common.Driver @DP.Person driverId
+  case entityRole of
+    DP.FLEET_OWNER -> do
+      DCommon.checkFleetOwnerVerification entityId merchant.fleetOwnerEnabledCheck
+      validateFleetOnwerWithDriver personId entityId merchant.id merchantOpCityId
+      removeDriverDocument personId
+    DP.OPERATOR -> do
+      validateOperatorWithDriver personId entityId
+      removeDriverDocument personId
+    _ -> throwError (InvalidRequest "Invalid Data")
+  pure Success
+  where
+    validateFleetOnwerWithDriver :: Id DP.Person -> Text -> Id DM.Merchant -> Id DMOC.MerchantOperatingCity -> Flow ()
+    validateFleetOnwerWithDriver personId fleetOwnerId merchantId merchantOpCityId = do
+      driver <- B.runInReplica $ QPerson.findById personId >>= fromMaybeM (PersonDoesNotExist personId.getId)
+      isFleetDriver <- FDV.findByDriverIdAndFleetOwnerId personId fleetOwnerId True
+      when (isNothing isFleetDriver) $ throwError DriverNotPartOfFleet
+      unless (merchantId == driver.merchantId && merchantOpCityId == driver.merchantOperatingCityId) $
+        throwError (PersonDoesNotExist personId.getId)
+    validateOperatorWithDriver :: Id DP.Person -> Text -> Flow ()
+    validateOperatorWithDriver personId operatorId = do
+      isDriverOperator <- DDriver.checkDriverOperatorAssociation personId (Id operatorId)
```

Comment:
Looks like indirect association operator <-> fleet + fllet <-> driver  did not handled here
See `checkFleetDriverAndDriverOperatorAssociation` function from this PR https://github.com/nammayatri/nammayatri/pull/11444/files